### PR TITLE
[Jest] Migrate `popup.test.js`

### PR DESCRIPTION
**Pull request by wipfli**
- Move and rename
- Run jscodeshift
- Run custom codemods and lint
- Manual edits

## Launch Checklist

 - [x] Confirm **your changes do not include backports from Mapbox projects** (unless with compliant license) - if you are not sure about this, please ask!
 - [ ] Briefly describe the changes in this PR.
 - [ ] Include before/after visuals or gifs if this PR includes visual changes.
 - [ ] Write tests for all new functionality.
 - [ ] Document any changes to public APIs.
 - [ ] Post benchmark scores.
 - [ ] Manually test the debug page.
 - [ ] Suggest a changelog category: bug/feature/docs/etc. or "skip changelog".
 - [ ] Add an entry inside this element for inclusion in the `maplibre-gl-js` changelog: `<changelog></changelog>`.

### CODE REVIEW

**Comment by wipfli** [comment]
Tests that do things like `expect(window.document.activeElement).toBe(dummyFocusedEl);` currently fail.

**Comment by wipfli** [comment]
Is jest-dom (https://www.npmjs.com/package/@testing-library/jest-dom) what we want to use here?

**Comment by wipfli** [comment]
> Is jest-dom (https://www.npmjs.com/package/@testing-library/jest-dom) what we want to use here?

No, but the element has to be added to `body` manually. See https://github.com/jsdom/jsdom/issues/2586#issuecomment-742593116

The following works:

```js
test('Popup with disabled focusing does not change the active element', () => {
    const dummyFocusedEl = window.document.createElement('button');
    window.document.body.appendChild(dummyFocusedEl);

    dummyFocusedEl.focus();

    new Popup({closeButton: false, focusAfterOpen: false})
        .setHTML('<span tabindex="0" data-testid="abc">Test</span>')
        .setLngLat([0, 0])
        .addTo(createMap());

    expect(window.document.activeElement).toBe(dummyFocusedEl);
});
```

File: src/ui/popup.test.ts [hunk 1]
```diff
@@ -0,0 +1,739 @@
+import {createMap as globalCreateMap, setPerformance, setWebGlContext} from '../util/test/util';
+import Popup, {Offset} from '../ui/popup';
```
Comment (HarelM) [inline src/ui/popup.test.ts:2]: Nit: shorter path.

**Review by HarelM — APPROVED** [review]
Fix the path and you are good to go :-)

**Comment by wipfli** [comment]
Thanks for reviewing @HarelM!

**Pull request merged**